packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -36,23 +38,20 @@ export const useGridSelection = (apiRef: GridApiRef): void => {
       !options.disableMultipleSelection && isMultipleKeyPressed;
   }, [isMultipleKeyPressed, options.disableMultipleSelection]);
 
-  const getSelectedRows = React.useCallback((): GridRowModel[] => {
-    // TODO replace with selector
-    return Object.keys(gridState.selection).map((id) => apiRef.current.getRowFromId(id));
-  }, [apiRef, gridState.selection]);
+  const getSelectedRows = React.useCallback(() => selectedRows, [selectedRows]);
```

Comment:
This might create a tricky bug!
Ie.
User select a row, event is selection change is published.
User subscribe to the event and call `getSelectedRows`.
As the grid did not rerender at that point, `selectedRows` is still set to the previous state. 🤔 
The user should use the params, with latest state but in case they don't...
Maybe you could add a test to check the above.

Alternatively you can make sure its' the latest state by 
```tsx
const getSelectedRows = React.useCallback(() => selectedGridRowsSelector(apiRef.current.getState()), []);
```